client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -93,87 +104,95 @@ const DomainPicker: FunctionComponent< Props > = ( {
-		}
-		if ( ! suggestion.is_free && numberOfPaidDomains < 5 ) {
-			numberOfPaidDomains++;
-			return suggestion;
-		}
-	} );
-
-	const handleDomainClick = ( suggestion: DomainSuggestion ) => {
-		if ( suggestion.is_free ) {
-			onDomainSelect( suggestion );
-		} else {
-			onDomainPurchase( suggestion );
-		}
-	};
+	const freeSuggestions = allSuggestions?.filter( suggestion => suggestion.is_free );
+	const paidSuggestions = allSuggestions?.filter( suggestion => ! suggestion.is_free );
+
+	// Recommend paid domain with exact site title match with highest relevance score.
+	const recommendedSuggestion = paidSuggestions?.find( suggestion =>
+		suggestion.match_reasons.includes( 'exact-match' )
+	);
 
 	return (
 		<Panel className="domain-picker">
 			<PanelBody>
 				<PanelRow className="domain-picker__panel-row">
-					<div className="domain-picker__choose-domain-header">
-						{ NO__( 'Choose a new domain' ) }
+					<div className="domain-picker__header">
+						<div className="domain-picker__header-title">{ NO__( 'Choose a new domain' ) }</div>
+						<Button className="domain-picker__close-button" isTertiary onClick={ () => onClose() }>
+							{ NO__( 'Skip for now' ) }
+						</Button>
+					</div>
+					<div className="domain-picker__search">
+						<SearchCard
+							className="domain-picker__search-field"
+							inputLabel={ label }
+							placeholder={ label }
+							onSearch={ setDomainSearch }
+							value={ domainSearch }
+							compact
+							hideClose
+							disableAutocorrect
+						/>
+						<div className="domain-picker__connect-domain">
+							<span>{ NO__( 'Already have a domain?' ) }</span>&nbsp;
+							<Button
+								className="domain-picker__connect-button"
+								isLink
+								onClick={ () => onDomainConnect() }
+							>
+								{ NO__( 'Connect it now' ) }
+							</Button>
+						</div>
 					</div>
-					<TextControl
-						hideLabelFromVision
-						label={ label }
-						placeholder={ label }
-						onChange={ setDomainSearch }
-						value={ domainSearch }
-					/>
 				</PanelRow>
 
-				<HorizontalRule className="domain-picker__divider" />
+				<PanelRow className="domain-picker__panel-row">
+					<div className="domain-picker__suggestion-header">
+						<div className="domain-picker__suggestion-header-title">
+							{ NO__( 'Professional Domain' ) }
+						</div>
+						<div className="domain-picker__suggestion-header-description">
+							{ __experimentalCreateInterpolateElement(
+								NO__( '<Price>Free</Price> for the first year with a paid plan' ),
+								{ Price: <em /> }
+							) }
+						</div>
+					</div>
+					<div className="domain-picker__suggestion-item-group">
+						{ paidSuggestions?.length
+							? paidSuggestions.map( suggestion => (
+									<DomainPickerSuggestionItem
+										suggestion={ suggestion }
+										isRecommended={ suggestion === recommendedSuggestion }
+										isCurrent={ currentDomain.domain_name === suggestion.domain_name }
```

Comment:
This is a type error, isn't it?

```suggestion
										isCurrent={ currentDomain?.domain_name === suggestion.domain_name }
```